src/components/frontend-engine/yup/helper.ts

```diff
@@ -16,6 +16,12 @@ interface IYupCombinedRule extends IYupRenderRule, IYupValidationRule {}
 export namespace YupHelper {
 	const customYupConditions: string[] = [];
 
+	// Yup's escape hatch for cycling dependency error
+	// this happens when 2 fields have conditional validation that rely on each other
+	// typically used to ensure user fill in one of many fields
+	// https://github.com/jquense/yup/issues/176#issuecomment-367352042
+	const whenPairIds: [string, string][] = [];
```

Comment:
- have moved `whenPairIds` into `parseWhenKeys()`

> just to confirm we only need to cater for pairs because we evaulate each key in the `when` rule one at a time?
- we cater to pairs only because that is what the `shape()` method accepts
- under-the-hood, Yup merely needs to know which field to process first so specifying pairs work
- say there is a three-way conditional validation where all three fields depend on each other, `parseWhenKeys` will return the pairing of all permutations of the three fields and Yup will parse the fields according to the pairings given